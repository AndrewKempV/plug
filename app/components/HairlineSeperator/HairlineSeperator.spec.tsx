import { StyleProp, ViewStyle } from "react-native";
import renderer from "react-test-renderer";
import Metrics from "../../config/metrics";
import mockRNLocalization from "../../config/mocks/react-native-localization-mock";
import HairlineSeparatorWithText from "./HairlineSeperator";

jest.mock("react-native-localization", () => mockRNLocalization);
const hairLineStyle = {
  style: { marginBottom: Metrics.margin2x * 3 } as StyleProp<ViewStyle>
};

test("Hairline seperator renders correctly", () => {
  const tree = renderer
    .create(HairlineSeparatorWithText(hairLineStyle))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
