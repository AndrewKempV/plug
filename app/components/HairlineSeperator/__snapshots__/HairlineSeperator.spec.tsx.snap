// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hairline seperator renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginHorizontal": 70,
      },
      Object {
        "marginBottom": 60,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4e5052",
        "height": 1,
        "marginTop": 10,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#4e5052",
        "fontFamily": "HelveticaNeue",
        "fontSize": 14,
        "fontWeight": "normal",
        "paddingBottom": 5,
        "paddingHorizontal": 5,
      }
    }
  >
    or
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#4e5052",
        "height": 1,
        "marginTop": 10,
        "width": 96,
      }
    }
  />
</View>
`;
